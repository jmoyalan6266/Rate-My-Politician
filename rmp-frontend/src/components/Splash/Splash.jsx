import { Jumbotron, Button, Card, CardDeck, Container } from "react-bootstrap";
import { Link } from "react-router-dom";
import background from "./background1.jpeg";
import useSingleSource from "../../hooks/useSingleSource";

const Splash = () => {
    const hStyle = {
        color: "black",
        fontFamily: "courier",
        textAlign: "left",
        fontSize: "350%",
    };
    const hStyle1 = {
        color: "black",
        fontFamily: "courier",
        textAlign: "center",
        fontSize: "300%",
    };
    const msnbc = useSingleSource(74);
    const wsj = useSingleSource(93);
    const dp = useSingleSource(43);

    return (
        <div>
            <div
                style={{
                    backgroundImage: `url(${background})`,
                    height: "100vh",
                    backgroundPosition: "center",
                    backgroundRepeat: "no-repeat",
                    backgroundSize: "cover",
                }}
            >
                <br></br>
                <h1 style={hStyle}>&nbsp; The REAL Truth</h1>
                <h1 style={hStyle}>&nbsp; Behind Your Politician.</h1>
                <br></br>
                &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
                <Button variant="dark" size="lg" as={Link} to="/states">
                    Find your State
                </Button>{" "}
                &nbsp; &nbsp; &nbsp; &nbsp;
                <Button variant="dark" size="lg" as={Link} to="/politicians">
                    Find your Politician
                </Button>
            </div>
            <br></br>
            <h1 style={hStyle1}>Popular News Sources to Check Out</h1>
            <Container>
                <CardDeck>
                    <Card>
                        <Card.Img variant="top" src={msnbc.source.logo} />
                        <Card.Body>
                            <center>
                                <Card.Title>{msnbc.source.name}</Card.Title>
                                <Card.Text>
                                    <strong>Based in:</strong> {msnbc.source.city}
                                    {", "} {msnbc.source.state}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Year Founded:</strong>{" "}
                                    {msnbc.source.foundedYear}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Number of Employees:</strong>{" "}
                                    {msnbc.source.employees}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Twitter:</strong> {msnbc.source.twitter}
                                </Card.Text>
                                <Button
                                    variant="dark"
                                    as={Link}
                                    to={`/news/${msnbc.source.index}`}
                                >
                                    More Info
                                </Button>
                            </center>
                        </Card.Body>
                    </Card>
                    <Card>
                        <Card.Img variant="top" src={wsj.source.logo} />
                        <Card.Body>
                            <center>
                                <Card.Title>{wsj.source.name}</Card.Title>
                                <Card.Text>
                                    <strong>Based in:</strong> {wsj.source.city}
                                    {", "} {wsj.source.state}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Year Founded:</strong>{" "}
                                    {wsj.source.foundedYear}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Number of Employees:</strong>{" "}
                                    {wsj.source.employees}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Twitter:</strong> {wsj.source.twitter}
                                </Card.Text>
                                <Button
                                    variant="dark"
                                    as={Link}
                                    to={`/news/${wsj.source.index}`}
                                >
                                    More Info
                                </Button>
                            </center>
                        </Card.Body>
                    </Card>
                    <Card>
                        <Card.Img variant="top" src={dp.source.logo} />
                        <Card.Body>
                            <center>
                                <Card.Title>{dp.source.name}</Card.Title>
                                <Card.Text>
                                    <strong>Based in:</strong> {dp.source.city}
                                    {", "} {dp.source.state}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Year Founded:</strong>{" "}
                                    {dp.source.foundedYear}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Number of Employees:</strong>{" "}
                                    {dp.source.employees}
                                </Card.Text>
                                <Card.Text>
                                    <strong>Twitter:</strong> {dp.source.twitter}
                                </Card.Text>
                                <Button
                                    variant="dark"
                                    as={Link}
                                    to={`/news/${dp.source.index}`}
                                >
                                    More Info
                                </Button>
                            </center>
                        </Card.Body>
                    </Card>
                </CardDeck>
            </Container>
            <br></br>
            <Jumbotron>
                <center>
                    <h1>How well do you know your politician?</h1>
                    <p>
                        In this world filled with an abundance of information, how
                        do we know what to believe and what information is accurate?
                        We hear a lot about fake news and a lot of differing stories
                        about the same people. Our website hopes to shed some light
                        on this and begin to see how different news outlets cover
                        politicians.
                    </p>
                    <p>
                        <Button variant="dark" as={Link} to="/about">
                            Learn more
                        </Button>
                    </p>
                </center>
            </Jumbotron>
        </div>
    );
};

export default Splash;
