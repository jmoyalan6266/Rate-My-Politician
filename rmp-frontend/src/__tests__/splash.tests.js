import React from "react"
import { configure, shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Splash from "../components/Splash/Splash"

configure({ adapter: new Adapter() })

describe("Test page views", () => {
    test("Splash", () => {
        const splashTest = shallow(<Splash />)
        expect(splashTest).toMatchSnapshot()
    })
})
