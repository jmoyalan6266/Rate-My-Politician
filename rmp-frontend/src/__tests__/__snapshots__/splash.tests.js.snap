// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page views Splash 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundImage": "url(background1.jpeg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "height": "100vh",
      }
    }
  >
    <br />
    <h1
      style={
        Object {
          "color": "black",
          "fontFamily": "courier",
          "fontSize": "350%",
          "textAlign": "left",
        }
      }
    >
        The REAL Truth
    </h1>
    <h1
      style={
        Object {
          "color": "black",
          "fontFamily": "courier",
          "fontSize": "350%",
          "textAlign": "left",
        }
      }
    >
        Behind Your Politician.
    </h1>
    <br />
                   
    <Button
      active={false}
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      disabled={false}
      size="lg"
      to="/states"
      variant="dark"
    >
      Find your State
    </Button>
     
           
    <Button
      active={false}
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      disabled={false}
      size="lg"
      to="/politicians"
      variant="dark"
    >
      Find your Politician
    </Button>
  </div>
  <br />
  <h1
    style={
      Object {
        "color": "black",
        "fontFamily": "courier",
        "fontSize": "300%",
        "textAlign": "center",
      }
    }
  >
    Popular News Sources to Check Out
  </h1>
  <Container
    fluid={false}
  >
    <CardDeck>
      <Card
        body={false}
      >
        <CardImg
          variant="top"
        />
        <CardBody>
          <center>
            <CardTitle />
            <CardText>
              <strong>
                Based in:
              </strong>
               
              , 
               
            </CardText>
            <CardText>
              <strong>
                Year Founded:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Number of Employees:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Twitter:
              </strong>
               
            </CardText>
            <Button
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              disabled={false}
              to="/news/undefined"
              variant="dark"
            >
              More Info
            </Button>
          </center>
        </CardBody>
      </Card>
      <Card
        body={false}
      >
        <CardImg
          variant="top"
        />
        <CardBody>
          <center>
            <CardTitle />
            <CardText>
              <strong>
                Based in:
              </strong>
               
              , 
               
            </CardText>
            <CardText>
              <strong>
                Year Founded:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Number of Employees:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Twitter:
              </strong>
               
            </CardText>
            <Button
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              disabled={false}
              to="/news/undefined"
              variant="dark"
            >
              More Info
            </Button>
          </center>
        </CardBody>
      </Card>
      <Card
        body={false}
      >
        <CardImg
          variant="top"
        />
        <CardBody>
          <center>
            <CardTitle />
            <CardText>
              <strong>
                Based in:
              </strong>
               
              , 
               
            </CardText>
            <CardText>
              <strong>
                Year Founded:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Number of Employees:
              </strong>
               
            </CardText>
            <CardText>
              <strong>
                Twitter:
              </strong>
               
            </CardText>
            <Button
              active={false}
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              disabled={false}
              to="/news/undefined"
              variant="dark"
            >
              More Info
            </Button>
          </center>
        </CardBody>
      </Card>
    </CardDeck>
  </Container>
  <br />
  <Jumbotron
    fluid={false}
  >
    <center>
      <h1>
        How well do you know your politician?
      </h1>
      <p>
        In this world filled with an abundance of information, how do we know what to believe and what information is accurate? We hear a lot about fake news and a lot of differing stories about the same people. Our website hopes to shed some light on this and begin to see how different news outlets cover politicians.
      </p>
      <p>
        <Button
          active={false}
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          disabled={false}
          to="/about"
          variant="dark"
        >
          Learn more
        </Button>
      </p>
    </center>
  </Jumbotron>
</div>
`;
